en/debug/profiling.md

```diff
@@ -12,6 +13,40 @@ The folded stacks are then fed into the visualization script, for which purpose
 
 ## Basic Usage
 
+### Prerequisites
+
+The profiler relies on GDB to run PX4 on the embedded target.
+So before profiling a target, you must have the hardware you wish to profile, and you must compile and upload the firmware to that hardware.
+You will then need a [SWD (JTAG) Hardware Debugging Interface](../debug/swd_debug.md#debug-probes), such as the DroneCode Probe, to run the GDB server and interact with the board.
+
+
+### Determine the Debugger Device
+
+The `poor-mans-profiler.sh` automatically detects and uses the correct USB device if you use it with a [DroneCode Probe](../debug/swd_debug.md#dronecode-probe).
+If you use a different kind of probe you may need to pass in the specific _device_ on which the debugger is located.
+You can use the bash command `ls -alh /dev/serial/by-id/` to enumerate the possible devices on Ubuntu.
+For example the following devices are enumerated with a Pixhawk 4 and DroneCode Probe connected over USB:
+```
+user@ubuntu:~/PX4-Autopilot$ ls -alh /dev/serial/by-id/
+total 0
+drwxr-xr-x 2 root root 100 Apr 23 18:57 .
+drwxr-xr-x 4 root root  80 Apr 23 18:48 ..
+lrwxrwxrwx 1 root root  13 Apr 23 18:48 usb-3D_Robotics_PX4_FMU_v5.x_0-if00 -> ../../ttyACM0
+lrwxrwxrwx 1 root root  13 Apr 23 18:57 usb-Black_Sphere_Technologies_Black_Magic_Probe_BFCCB401-if00 -> ../../ttyACM1
+lrwxrwxrwx 1 root root  13 Apr 23 18:57 usb-Black_Sphere_Technologies_Black_Magic_Probe_BFCCB401-if02 -> ../../ttyACM2
+```
+
+In this case, the script would automatically pick up the device named `*Black_Magic_Probe*-if00`.
+But if you were using a different device you would be able discover the appropriate id from the listing above. 
+
+Then pass in the appropriate device using the `--gdbdev` argument like this:
+```bash
+./poor-mans-profiler.sh --elf=build/px4_fmu-v4_default/px4_fmu-v4_default.elf --nsamples=30000 --gdbdev=/dev/ttyACM2
```

Comment:
@JacobCrabill Just to confirm, is this correct format `--gdbdev=/dev/ttyACM2` ?

Reply:
Yes that would be correct.  I didn't specifically mention that option here previously because most people will likely rely on the `.cmake`-defined `profile` build option, which calls the script and defines the command-line options for them.

Reply:
@JacobCrabill So would you/could you then normally update the profile build option, and if so, where/how?

Reply:
The command you listed would work just fine.